src/button-dropdown/item-element/index.tsx

```diff
@@ -23,6 +23,7 @@ const ItemElement = ({
   first = false,
   last,
   hasCategoryHeader,
+  focused = false,
```

Comment:
Why not have the `highlightedType` property same to select/multiselect? The `focused` flag is not very clear in the context of the item-element component. If the item actually receives focus then in can be checked by comparing it against the document.activeElement. If the `focused` is only referring to the difference in visual highlight, I'd at least rename it to e.g. `isKeyboardHighlight`.

Reply:
I think it is logically focused, because isFocus() checks if the item is last the one currently highlight (because we also highlight the parent item of the highlighted one) and if it is by keyboard (mouse is hover) -- it is what we expect from focus, or not? Also one of purposes of this change is to differentiate focus style and hover style. We can't use document.activeElement because highlighted item is always focus() but from user view mouse highlight is hover https://github.com/cloudscape-design/components/blob/main/src/button-dropdown/item-element/index.tsx#L78

Reply:
So if the item is always focused, having that `focused` property is a little confusing. Would it make sense to at least rename it or document with a comment?

Reply:
renamed to isKeyboardHighlight and isKeyboardHighlighted and also added comment to isKeyboardHighlight function.